mesh/pmesh.cpp

```diff
@@ -5568,6 +5568,97 @@ void ParMesh::PrintSharedEntities(const char *fname_prefix) const
    }
 }
 
+void ParMesh::GetGlobalVertexIndices(Array<HYPRE_Int> &gi)
+{
+   H1_FECollection fec(1, Dim); // Order 1, mesh dimension (not spatial dimension).
+   ParFiniteElementSpace fespace(this, &fec);
+
+   MFEM_VERIFY(fespace.GetVSize() == GetNV(),
+               "There should be a bijection between DOFs and vertices");
+
+   gi.SetSize(GetNV());
+
+   Array<int> dofs;
+   bool oneDof = true;
+   for (int i=0; i<GetNV(); ++i)
+   {
+      fespace.GetVertexDofs(i, dofs);
+      oneDof = oneDof && (dofs.Size() == 1);
+      gi[i] = fespace.GetGlobalTDofNumber(dofs[0]);
+   }
+
+   MFEM_VERIFY(oneDof, "Each vertex should have exactly one DOF");
```

Comment:
Same question here. Is this check really necessary?

Reply:
This check is not necessary as it is fully controlled by the local objects `fespace` and `fec`. If `H1_FECollection fec(1, Dim); ParFiniteElementSpace fespace(this, &fec);` starts to give more than one dof per vertex for some reason in the future, this function will be the least of our problems.